doc/specs/stdlib_linalg.md

```diff
@@ -206,3 +206,155 @@ program demo_outer_product
     !A = reshape([3., 6., 9., 4., 8., 12.], [3,2])
 end program demo_outer_product
 ```
+
+## `zeros/ones` - Create a `vector` or `matrix` of the given shape, filled completely with either `0` or `1` `integer [/real/complex]` type values
+
+### Status
+
+Experimental
+
+### Class
+
+Pure function.
+
+### Description
+
+`zeros` creates a `vector` or `matrix` of the given shape, filled completely with `0` `integer [/real/complex]` type values.  
+`ones` creates a `vector` or `matrix` of the given shape, filled completely with `1` `integer [/real/complex]` type values.
+
+### Syntax
+
+For vector:  
+`result = [[stdlib_linalg(module):zeros(interface)]](dim)`  
+`result = [[stdlib_linalg(module):ones(interface)]](dim)`
+
+For matrix:  
+`result = [[stdlib_linalg(module):zeros(interface)]](dim1, dim2)`  
+`result = [[stdlib_linalg(module):ones(interface)]](dim1, dim2)`
+
+
+### Arguments
+
+`dim/dim1`: Shall be an `integer` type.
+This is an `intent(in)` argument.
+
+`dim2`: Shall be an `integer` type.
+This is an `intent(in)` argument.
+
+#### Note
+
+Because of `huge(integer :: i) == 2147483647`, the dimensional maximum length of array created by the `zeros/ones` function is `2147483647`. 
+
+### Return value
+
+Return a `vector` or `matrix` of the given shape, filled completely with either `0` or `1` `integer [/real/complex]` type values.
+
+#### Warning
+
+If the array that receives the return value of the `zeros/ones` function is of `real/complex` type, conversion from `integer` type to `real/complex` type will occur.  
+
+Just as `Fortran` is a strongly typed statically compiled language, be careful with the following statements:
+```fortran
+real :: A(:,:)
+
+!> Be careful
+A = ones(2,2)/2     !! A = 1/2 = 0.0
+
+!> Recommend
+A = ones(2,2)/2.0   !! A = 1/2.0 = 0.5
+A = expand(0.5, 2,2)
+```
+
+### Example
+
+```fortran
+program demo
+    use stdlib_linalg, only: zeros, ones
+    implicit none
+    real, allocatable :: A(:,:)
+    integer :: iA(2)
+    compelx :: cA(2), cB(2,3)
+    
+    A = zeros(2,2)          !! [0.0,0.0; 0.0,0.0]
+    A = ones(4,4)           !! [1.0,1.0,1.0,1.0; 1.0,1.0,1.0,1.0; 1.0,1.0,1.0,1.0; 1.0,1.0,1.0 1.0]
+    A = 2.0*ones(2,2)       !! [2.0,2.0; 2.0,2.0]
+
+    iA = ones(2)            !! [1,1]
+    cA = ones(2)            !! [(1.0,0.0),(1.0,0.0)]
+    cA = (1.0,1.0)*ones(2)  !! [(1.0,1.0),(1.0,1.0)]
+    cB = ones(2,3)          !! [(1.0,0.0),(1.0,0.0),(1.0,0.0); (1.0,0.0),(1.0,0.0),(1.0,0.0)]
+
+end program demo
+```
+
+## `expand` - Create a `vector` or `matrix` of the given shape, filled completely with `value` `integer/logical/real/complex/string_type` type values
+
+### Status
+
+Experimental
+
+### Class
+
+Pure function.
+
+### Description
+
+`expand` creates a `vector` or `matrix` of the given shape, filled completely with `value` `integer/logical/real/complex/string_type` type values.
+
+### Syntax
+
+For vector:  
+`result = [[stdlib_linalg(module):expand(interface)]](value, dim)`  
```

Comment:
`expand` function for vector in vanilla form may not be necessary since we already have `spread` intrinsic function for the same results. However, it may be interesting to have a third optional logical vector mask with size of `dim`.

Reply:
Yes, I was thinking of removing `expand`. The mask scheme you mentioned seems to be impossible to implement in `Fortran`'s existing syntax. We can't arbitrarily set the dimensions of the returned array in Fortran. (see https://github.com/j3-fortran/fortran_proposals/issues/157)
```fortran
x = reshepe(ones(2*3*4), [2,3,4])  !! This use style will be written in doc later.
```

Reply:
My suggest is like the following:
`function expand(value, dim, mask) result(res)`
`integer, intent(in) :: value, dim`
`logical, intent(in) :: mask(dim)`
`integer, allocatable :: res(:)`

`allocate(res(dim))`
`where(mask)`
`res = value`
`elsewhere`
`res = 0`
`end where`
`end function expand`

Reply:
It seems that `merge` may achieve this effect, for example:
```fortran
x = merge(1, 0, mask) !! `x` is consistent with the dimension information of the `mask` array.
```